src/unix/linux_like/linux/musl/mod.rs

```diff
@@ -415,6 +456,40 @@ s! {
         #[cfg(target_arch = "loongarch64")]
         pub tcpi_snd_wnd: u32,
     }
+
+    // MIPS implementation is special (see mips arch folders)
+    #[cfg(not(target_arch = "mips"))]
```

Comment:
The `x32` target is not the intel/AMD 32-bit target (that's in `i386` in musl), it's this https://sites.google.com/site/x32abi/.  Rust supports `gnux32` but doesn't support `muslx32` (yet?) so we don't need to worry about this weird architecture.  The padding fields are necessary on the x32 target since we're passing this structure to a 64-bit kernel so we need to shuffle the 32-bit user-space fields into the appropriate places in the kernel structure.

Reply:
Makes sense, thanks for the clarification!